src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets

```diff
@@ -17,11 +17,48 @@
     <Warning Condition="Exists('$(SerializerDllImmediatePath)') != 'true' And Exists('$(SerializerCsImmediatePath)') == 'true'" Text="$(SGenWarningText)"/>
     <Copy Condition="Exists('$(SerializerDllImmediatePath)') == 'true'" SourceFiles="$(SerializerDllImmediatePath)" DestinationFolder="$(OutputPath)" />
   </Target>
+  
   <Target Name="CleanSerializationAssembly" AfterTargets="CoreClean">
     <Message Text="Cleaning serialization files..." Importance="normal"/>
     <Delete Condition="Exists('$(OutputPath)\$(SerializerName).dll') == 'true'" Files="$(OutputPath)\$(SerializerName).dll" />
   </Target>
+  
   <Target Name="CopySerializer" AfterTargets="PrepareForPublish">
     <Copy Condition="Exists('$(OutputPath)\$(AssemblyName).XmlSerializers.dll')=='true'" SourceFiles="$(OutputPath)\$(AssemblyName).XmlSerializers.dll" DestinationFolder="$(PublishDir)" SkipUnchangedFiles="false" />
   </Target>
+  
+ <Target Name="GenerateSerializationAssemblyForReferenceAssembly" AfterTargets="GenerateSerializationAssembly" Condition="@(SerializationAssembly)!=''">
+    <ItemGroup Condition="@(SerializationAssembly)!=''" Label="Parse SerializationAssembly">
+      <SearchSerializationAssembly Include="@(Reference)">
+        <SerializationAssembly>%(SerializationAssembly.Identity)</SerializationAssembly>
+      </SearchSerializationAssembly>
+      <TargetSerializationAssembly Include="@(SearchSerializationAssembly)" Condition="$([System.String]::new('%(SearchSerializationAssembly.Identity)').EndsWith('%(SearchSerializationAssembly.SerializationAssembly).dll'))" />
+      <TargetSerializationAssemblyAndType Include="@(TargetSerializationAssembly)">
+        <TargetSerializationAssembly>%(SerializationAssembly.Identity)</TargetSerializationAssembly>
+        <TargetSerializationType>@(SerializationAssembly-&gt;Metadata('SerializationType'))</TargetSerializationType>
+      </TargetSerializationAssemblyAndType>
+    </ItemGroup>
+
+    <ItemGroup>
+      <ReferenceSerializerName Include="%(SerializationAssembly.Identity).XmlSerializers" />
+      <ReferenceSerializerDllImmediatePath Include="$(IntermediateOutputPath)%(ReferenceSerializerName.Identity).dll" />
+      <ReferenceSerializerPdbImmediatePath Include="$(IntermediateOutputPath)%(ReferenceSerializerName.Identity).pdb" />
+      <ReferenceSerializerCsImmediatePath Include="$(IntermediateOutputPath)%(ReferenceSerializerName.Identity).cs" />
```

Comment:
I think we can keep only one `ReferenceSerializerImmediatePath`. It's not easy to distinguish the above 3 items. Besides, it seems in some case we cannot use them.